endpoints/openrtb2/amp_auction.go

```diff
@@ -162,17 +163,28 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 			}
 		}
 	}
+	// Extract any errors
+	var extResponse openrtb_ext.ExtBidResponse
+	var respErrors map[openrtb_ext.BidderName][]openrtb_ext.ExtBidderError
+	eRErr := json.Unmarshal(response.Ext, &extResponse)
+	if eRErr != nil {
+		glog.Errorf("AMP response: failed to unpack OpenRTB response.ext: %v", eRErr)
+		ao.Errors = append(ao.Errors, fmt.Errorf("AMP response: failed to unpack OpenRTB response.ext: %v", eRErr))
+	} else {
+		respErrors = extResponse.Errors
+	}
+
 	// Now JSONify the targets for the AMP response.
 	ampResponse := AmpResponse{
 		Targeting: targets,
+		Errors:    respErrors,
 	}
 
 	ao.AmpTargetingValues = targets
 
 	// add debug information if requested
-	if req.Test == 1 {
-		var extResponse openrtb_ext.ExtBidResponse
-		if err := json.Unmarshal(response.Ext, &extResponse); err == nil && extResponse.Debug != nil {
+	if req.Test == 1 && eRErr == nil {
```

Comment:
why?

Reply:
If we had an error above in unmarshalling the ext, we have already logged it and then this message could be deceptive (its not that errors was missing, but that the whole ext was damaged). So test for both if debug was set, no errors unpacking, and we have debug info.